### PR TITLE
include the report upload runner in package

### PR DESCRIPTION
**Pull request by bochi**
Include runner and reactor for the salt report upload in /usr/share/foreman-salt 
This allows for easier configuration and especially updating the runner and reactor using the package. 

Salt configuration in this case would look like so:
```
runner_dirs:
  - /usr/share/foreman-salt/runner
```
and the reactor config:
```
reactor:
  - 'salt/job/*/ret/*':
    - /usr/share/foreman-salt/reactor/foreman_report_upload.sls
```

### CODE REVIEW

**Comment by sbernhard** [comment]
Should we then remove https://github.com/theforeman/foreman-packaging/blob/fc5c1f43cb755dff78945aac6f355c9a604778ac/packages/plugins/rubygem-smart_proxy_salt/rubygem-smart_proxy_salt.spec#L176 ?

**Comment by bochi** [comment]
> Should we then remove
> 
> https://github.com/theforeman/foreman-packaging/blob/fc5c1f43cb755dff78945aac6f355c9a604778ac/packages/plugins/rubygem-smart_proxy_salt/rubygem-smart_proxy_salt.spec#L176
> 
> ?

Ah, haven't noticed that its there, but yes, I would say so as gem_instdir isn't persistant across versions.

**Comment by sbernhard** [comment]
Ok, I guess the /salt should be removed if no longer required. Additionally, we need to update the documentation, again :)

**Comment by bochi** [comment]
> Ok, I guess the /salt should be removed if no longer required. Additionally, we need to update the documentation, again :)

Will do asap.

**Comment by sbernhard** [comment]
[test rpm]

**Comment by bochi** [comment]
> [test rpm]

Why are there merge conflicts? Trying it manually on the cli seems to merge fine.

**Comment by ekohl** [comment]
Git has more complex merge algorithms than Github in detecting conflicts. You should rebase and force push.

**Comment by bochi** [comment]
> Git has more complex merge algorithms than Github in detecting conflicts. You should rebase and force push.

I believe things should be OK now...

**Comment by ekohl** [comment]
No, you merged develop into it instead of rebasing.

**Comment by bochi** [comment]
> No, you merged develop into it instead of rebasing.

Geez, what a mess. I will try to resolve this tomorrow, or even create a new PR and close this one if the mess is too big. I don't even get where @sbernhard changes are coming from in the conflicts, they are not in rpm/develop

**Comment by bochi** [comment]
> > No, you merged develop into it instead of rebasing.
> 
> Geez, what a mess. I will try to resolve this tomorrow, or even create a new PR and close this one if the mess is too big. I don't even get where @sbernhard changes are coming from in the conflicts, they are not in rpm/develop

Sorry for the confusion. The changes were there upstream of course. 
Anyway, I hard reset the branch to rpm/develop and patched my changes back in. So finally we should be good, right?

**Comment by sbernhard** [comment]
[test rpm]

**Comment by bochi** [comment]
worth a try :laughing: @sbernhard  can you trigger the test again please, i fixed a wrong directoryi

File: packages/plugins/rubygem-smart_proxy_salt/rubygem-smart_proxy_salt.spec [hunk 1]
```diff
@@ -30,12 +30,15 @@
 %global foreman_proxy_settingsd_dir %{_root_sysconfdir}/foreman-proxy/settings.d
 %global smart_proxy_dynflow_bundlerd_dir %{_datadir}/smart_proxy_dynflow_core/bundler.d
 
+%global foreman_salt_datadir /usr/share/foreman-salt
```
Comment (sbernhard) [inline packages/plugins/rubygem-smart_proxy_salt/rubygem-smart_proxy_salt.spec:33]: I guess this should be %{_root_datadir}/foreman-salt/

Comment (ehelms) [reply packages/plugins/rubygem-smart_proxy_salt/rubygem-smart_proxy_salt.spec:33]: Is `foreman-salt` the right directory given this is a smart proxy plugin?

**Comment by bochi** [comment]
Good question I thought it would be a good choice. Do you think it should go somewhere under /usr/share/foreman-proxy even though the directory is managed by another package?

**Comment by pcreech** [comment]
Looks like this PR now has some merge conflicts. Can we get it updated?

**Comment by pcreech** [comment]
Closing due to inactivity.  If you feel it needs to be reopened, please reopen it!

**Pull request closed without merging**